### ISSUE TITLE
Exec to Pod via kubectl shell fails with error: unable to upgrade connection: Forbidden

### ISSUE THREAD

**Issue by ansilh** [rancher/rancher#29668]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Login as Admin user
- Deploy an app and try to connect to it via UI shell
```
> kubectl get pods -n default
NAME                     READY   STATUS    RESTARTS   AGE
myapp-6f588497df-vjdsg   1/1     Running   0          9m58s
> kubectl exec -it -n default myapp-6f588497df-vjdsg 
error: you must specify at least one command for the container
> kubectl exec -it -n default myapp-6f588497df-vjdsg -- bash
error: unable to upgrade connection: Forbidden
> 
```
**Result:**

Exec results in error `error: unable to upgrade connection: Forbidden`

**Other details that may be helpful:**

- Connection via UI shell works fine
- The same steps worked fine on a Rancher v2.4.4 environment 

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.5.1
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): NA
- Kubernetes version (use `kubectl version`):
```
> kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T11:56:40Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.2", GitCommit:"f5743093fd1c663cb0cbc89748f730662345d44d", GitTreeState:"clean", BuildDate:"2020-09-16T13:32:58Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
> 
```

- Docker version (use `docker version`):

```
Client: Docker Engine - Community
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 17:02:36 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       4484c46d9d
  Built:            Wed Sep 16 17:01:06 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.3.7
  GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

gz#12982

gz#13209

**Comment by rajeshneo** [comment]
+1 Facing same issue

**Comment by jiaqiluo** [comment]
The bug is reproduced in v2.5.1 HA.  
This happens to any pod in the cluster, installing an app is not required. For example, we can create a deployment with the Nginx image. 

It turns out we cannot `exec` into the pod via neither the builtin `launch kubectl` in the MCM UI nor  `shell` in the Dashboard.

 
![image](https://user-images.githubusercontent.com/6218999/98425286-d2711c00-2051-11eb-8722-fafc34665195.png)


![image](https://user-images.githubusercontent.com/6218999/98425299-e026a180-2051-11eb-81fd-f05f0200a0fb.png)

**Comment by jiaqiluo** [comment]
the bug also exits on `v2.5-715fcc719de073a58a21410b10cdff2da94acc1b-head`

**Comment by StrongMonkey** [comment]
Available to test in master

**Comment by jiaqiluo** [comment]
The bug fix is validated on rancher:master-head 4e5b1a5
Now we can `exec` to a pod in the builtin shell on both UIs.
  
<img width="766" alt="Screen Shot 2020-11-10 at 3 13 03 PM" src="https://user-images.githubusercontent.com/6218999/98739856-56cfe180-2367-11eb-86c2-4992c4d71bf2.png">

<img width="555" alt="Screen Shot 2020-11-10 at 3 12 37 PM" src="https://user-images.githubusercontent.com/6218999/98739852-55061e00-2367-11eb-9213-fe9372e0a175.png">

**Comment by jiaqiluo** [comment]
reopen for waiting for the fix in the v2.5 branch

**Comment by StrongMonkey** [comment]
Available to test on 2.5-head once build is done.

**Comment by jiaqiluo** [comment]
The bug fix is validated on rancher:`v2.5-head` 71da156
Now we can exec to a pod in the builtin shell on both UIs.

![image](https://user-images.githubusercontent.com/6218999/99110861-45294c80-25a8-11eb-88b9-67484ffb9fbc.png)

![image](https://user-images.githubusercontent.com/6218999/99111009-8e799c00-25a8-11eb-8030-d151946ea998.png)

**Comment by rajeshneo** [comment]
Facing same issue on 2.5.2 as well

**Comment by ansilh** [comment]
Hi @rajeshneo , The milestone for this fix is targetted for Rancher 2.5.4 (This may change if there is any other issue to be prioritized before the targetted milestone)

![image](https://user-images.githubusercontent.com/11064902/102451830-45ad7b00-405f-11eb-85a4-657bba2f8437.png)

**Comment by rajeshneo** [comment]
oh. sorry, I missed the milestone version. Since it was closed I assumed it is done. @ansilh Thanks for the update.
